File: 2Star AdvLvl/CLMBSTRS.py
MAXN = 1000000
MODN = 1000000007

table = [0] * (MAXN + 1)

def count_ones(x):
    count = 0
    while x:
        count = count + (x & 1)
        x >>= 1
    return count

def monkRitual():
    N, G = map(int, input().split())
    print("CORRECT" if count_ones(table[N]) == G else "INCORRECT")

def main():
    table[0] = 0
    table[1] = 1
    table[2] = 2
    for i in range(3, MAXN + 1):
        table[i] = (table[i - 1] % MODN + table[i - 2] % MODN) % MODN
    tt = int(input())
    for _ in range(tt):
        monkRitual()

if __name__ == "__main__":
    main()

File: 2Star AdvLvl/CLMBSTRS.cpp
#include <iostream>
using namespace std;
#define MAXN 1000000
#define MODN 1000000007

long long table[MAXN + 1];
long long count_ones(long long x) {
    long long count = 0;
    while (x) {
        count = count + (x & 1);
        x >>= 1;
    }
    return count;
}
void monkRitual() {
    int N, G;
    cin >> N >> G;
    cout << (count_ones(table[N]) == G ? "CORRECT" : "INCORRECT")<<'\n';
}
int main() {
    table[0] = 0;
    table[1] = 1;
    table[2] = 2;
    for (int i=3; i<=MAXN; i++) {
        table[i] = (table[i-1]%MODN + table[i-2]%MODN)%MODN;
    }
	int tt;
	cin >> tt;
	while (tt--) {
	    monkRitual();
	}
	return 0;
}
